### PR TITLE
Show room options menu if "UIComponent.roomOptionsMenu" is enabled

### PR DESCRIPTION
**Pull request by maheichyk**
Type: Enhancement

`"UIComponent.roomOptionsMenu"` is added as a new component that can be used by `shouldShowComponent`.

This PR suggests to enable/disable rendering of "Room options" menu in several places: `RoomHeader.tsx`, `RoomTile.tsx`, `RoomResultContextMenus.tsx`.

Some users may have very limited functionality, so in this case it would be great to hide room options completely.

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Show room options menu if "UIComponent.roomOptionsMenu" is enabled ([\#10365](https://github.com/matrix-org/matrix-react-sdk/pull/10365)). Contributed by @maheichyk.

### CODE REVIEW

**Review by t3chguy — APPROVED** [review]
Looks sane to me, sorry for this slipping the cracks

File: test/components/views/rooms/RoomTile-test.tsx [hunk 1]
```diff
@@ -285,7 +305,7 @@ describe("RoomTile", () => {
         });
 
         it("should render a room without a message as expected", async () => {
-            renderRoomTile();
+            const renderResult: RenderResult = renderRoomTile();
```
Comment (t3chguy) [inline test/components/views/rooms/RoomTile-test.tsx:308]: Can we ditch these explicit types? `renderRoomTile` has a clear return type, this just makes refactors more messy

File: test/components/views/rooms/RoomTile-test.tsx [hunk 2]
```diff
@@ -297,7 +317,7 @@ describe("RoomTile", () => {
             });
 
             it("should render as expected", async () => {
-                renderRoomTile();
+                const renderResult: RenderResult = renderRoomTile();
```
Comment (t3chguy) [inline test/components/views/rooms/RoomTile-test.tsx:320]: Same as above

File: test/components/views/rooms/RoomTile-test.tsx [hunk 3]
```diff
@@ -309,7 +329,7 @@ describe("RoomTile", () => {
...
```
Comment (t3chguy) [inline test/components/views/rooms/RoomTile-test.tsx:332]: Same as above

File: test/components/views/rooms/RoomTile-test.tsx [hunk 4]
```diff
@@ -148,12 +153,24 @@ describe("RoomTile", () => {
     });
 
     describe("when message previews are not enabled", () => {
-        beforeEach(() => {
+        it("should render the room", () => {
+            mocked(shouldShowComponent).mockReturnValue(true);
+            const renderResult: RenderResult = renderRoomTile();
```
Comment (t3chguy) [inline test/components/views/rooms/RoomTile-test.tsx:158]: Same as elsewhere

**Review by t3chguy — COMMENTED** [review]
Looks sane otherwise

File: test/components/views/rooms/RoomTile-test.tsx (hunk 1 above)
Comment (maheichyk) [reply test/components/views/rooms/RoomTile-test.tsx]: fixed, thanks for issue

File: test/components/views/rooms/RoomTile-test.tsx (hunk 2 above)
Comment (maheichyk) [reply test/components/views/rooms/RoomTile-test.tsx]: fixed there too

File: test/components/views/rooms/RoomTile-test.tsx (hunk 3 above)
Comment (maheichyk) [reply test/components/views/rooms/RoomTile-test.tsx]: fixed

File: test/components/views/rooms/RoomTile-test.tsx (hunk 4 above)
Comment (maheichyk) [reply test/components/views/rooms/RoomTile-test.tsx]: it is ok

**Review by t3chguy — APPROVED** [review]
Thanks!!

**Pull request merged**